idea/meta-storage/src/services/meta/meta.service.ts

```diff
@@ -0,0 +1,51 @@
+import { plainToClass } from 'class-transformer';
+import { HexString } from '@polkadot/util/types';
+import { getProgramMetadata } from '@gear-js/api';
+
+import { CreateMetaInput } from './input/create-meta.input';
+import { Meta } from '../../database/entities';
+import { metaRepository } from '../../database/repositories';
+import { MetaAlreadyExists, MetadataNotFound } from '../../common/errors';
+import { generateHash } from '../../common/helpers/generate-hash';
+
+export const metaService = {
+  async createByIndexer(createMetaInput: CreateMetaInput): Promise<Meta | null> {
+    const metaDB = await metaRepository.get(createMetaInput.hash);
+
+    if (metaDB) return;
+
+    const metaTypeDB = plainToClass(Meta, {
+      id: createMetaInput.hash,
+    });
+
+    return metaRepository.save(metaTypeDB);
+  },
+
+  async create(createMetaInput: CreateMetaInput): Promise<Meta> {
```

Comment:
I have separated methods because from where comes the request we should have different behaviors. What I mean by that:
1. every request that comes from `indexer` does not need to call an exception that some `meta` exist or not by hash, just needs to handle and also this request does not need to use a request and reply pattern.
2. when a request comes from `idea` need to call some exception if meta already exists, and of course, we should reply at the end of the process.